gitpod-io/gitpod
# [server] incremental workspaces

## Description
Introduces a project setting that allows for starting workspaces on the successful prebuild on the direct linear commit history. Enabling this will always skip the "prebuild-in-progress" view.
It also allows specifying the number of commits to skip in between prebuilds.

## Related Issue(s)

Fixes #12583

## How to test
- create a team & project
- go to the settings and enable 'incremental prebuilds' set the number of commits to 1
- start a workspace.
- push a commit (no config change!) and verify that no prebuild is running
- start a workspace and verify that it is based on the first prebuild
- push a commit (no config change!) and verify that a prebuild gets triggered
- push a commit that changes the init task in the gitpod.yml and verify that a prebuild gets triggered

## Release Notes

```release-note
Introduced a new project setting that allows starting workspaces based on the last successful prebuild in the git commit history.
```

## Documentation


## Werft options:

      If enabled this will build `install/preview`
      Valid options are `all`, `workspace`, `webapp`, `ide`
